kamon-core/src/main/scala/kamon/ActorSystemTools.scala

```diff
@@ -14,12 +14,12 @@
  */
 package kamon
 
-import scala.util.control.NonFatal
-
 import akka.actor.ActorSystem
 
 object ActorSystemTools {
   private[kamon] def terminateActorSystem(system: ActorSystem): Unit = {
-    system.shutdown()
+    // Use system.shutdown() if it exists (akka 2.4 and lower), system.terminate() otherwise (akka 2.5+)
+    system.getClass.getMethods.find(_.getName == "shutdown").map(_.invoke(system))
+      .getOrElse(system.getClass.getMethod("terminate").invoke(system))
```

Comment:
If you are getting rid of the scala-2.10/2.11/2.12 differences, could you move this back to src/main/scala/kamon ? I was just about to raise this as an issue myself.